environs/networking.go

```diff
@@ -118,6 +118,32 @@ type NetworkingEnviron interface {
 	Networking
 }
 
+// NoSpaceDiscoveryEnviron implements methods from Networking that represent an
+// environ without native space support (all but MAAS at the time of writing).
+// None of the method receiver references are used, so it can be embedded
+// as nil without fear of panics.
+type NoSpaceDiscoveryEnviron struct{}
+
+// SupportsSpaceDiscovery (Networking) indicates that
+// this environ does not support space discovery.
+func (*NoSpaceDiscoveryEnviron) SupportsSpaceDiscovery(context.ProviderCallContext) (bool, error) {
+	return false, nil
+}
+
+// Spaces (Networking) indicates that this provider
+// does not support returning spaces.
+func (*NoSpaceDiscoveryEnviron) Spaces(context.ProviderCallContext) (network.SpaceInfos, error) {
+	return nil, errors.NotSupportedf("Spaces")
+}
+
+// ProviderSpaceInfo (Networking) indicates that this provider
+// does not support returning provider info for the input space.
+func (*NoSpaceDiscoveryEnviron) ProviderSpaceInfo(
```

Comment:
not only should these not be called, but even if they are, they don't reference the receiver so won't panic/fail so definitely a nice fix